pkg/platform/model/buildplanner.go

```diff
@@ -544,9 +546,46 @@ func VersionStringToRequirements(version string) ([]bpModel.VersionRequirement,
 		}
 		parts[i-1] = strconv.Itoa(previousPart + 1)
 		requirements = append(requirements, bpModel.VersionRequirement{
-			bpModel.VersionRequirementComparatorKey: "lt",
+			bpModel.VersionRequirementComparatorKey: bpModel.ComparatorLT,
 			bpModel.VersionRequirementVersionKey:    strings.Join(parts[:i], "."),
 		})
 	}
 	return requirements, nil
 }
+
+func MonoModelConstraintsToRequirements(constraints *mono_models.Constraints) []*bpModel.VersionRequirement {
+	requirements := []*bpModel.VersionRequirement{}
+	for _, constraint := range *constraints {
+		requirements = append(requirements, &bpModel.VersionRequirement{
+			bpModel.VersionRequirementComparatorKey: constraint.Comparator,
+			bpModel.VersionRequirementVersionKey:    constraint.Version,
+		})
+	}
+	return requirements
+}
+
+func RequirementsToVersionString(requirements []*bpModel.VersionRequirement) string {
+	if requirements == nil || len(requirements) == 0 {
+		return ""
+	}
+
+	parts := make([]string, len(requirements))
+	for i, requirement := range requirements {
+		version := (*requirement)[bpModel.VersionRequirementVersionKey]
+		switch (*requirement)[bpModel.VersionRequirementComparatorKey] {
+		case bpModel.ComparatorEQ:
+			parts[i] = version
+		case bpModel.ComparatorGT:
+			parts[i] = fmt.Sprintf(">%s", version)
+		case bpModel.ComparatorGTE:
+			parts[i] = fmt.Sprintf(">=%s", version)
+		case bpModel.ComparatorLT:
+			parts[i] = fmt.Sprintf("<%s", version)
+		case bpModel.ComparatorLTE:
+			parts[i] = fmt.Sprintf("<=%s", version)
+		case bpModel.ComparatorNE:
+			parts[i] = fmt.Sprintf("!%s", version)
+		}
+	}
+	return strings.Join(parts, ",")
+}
```

Comment:
You should be able to drop this by using `VersionConstraint` (string) as mentioned in my other comment. I'd like to avoid encoding/decoding constraints on the client.

Reply:
I've removed this as requested.